lib/promql/expr.go

```diff
@@ -0,0 +1,423 @@
+package promql
```

Comment:
I split these definitions out from `parser.go` to have a more convenient place to understand all the node types in an expression.